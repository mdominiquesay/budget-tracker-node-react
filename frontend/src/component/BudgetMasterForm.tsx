
import { useState, useEffect,ChangeEvent } from 'react';
import { DialogDetails, Dialog } from '../Pages/Dialog';
import { FormData, SubmitBudgetMaster, defaultBudget} from "../Model/BudgetModel";
import BusyIndicator from '../component/BusyIndicator';
interface BudgetMasterFormProps {
    fetchData: () => void;
}

const BudgetMasterForm: React.FC<BudgetMasterFormProps> = ({ fetchData }) => {
    const [loading, setLoading] = useState<boolean>(false);
    const [details, setDetails] = useState<DialogDetails>({
        Button: "New Budget",
        Title: "Budget Name",
        Body: <h1>Body</h1>,
        Close: "Close",
        Save: "Save"
    });
    const [formData, setFormData] = useState<FormData>(defaultBudget);
    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        const { value } = event.target;
        const updatedFormData = { ...formData, budget_name: value };
        setFormData(updatedFormData);
    };
    const handleClose=  () => {
        setFormData(defaultBudget);
       }
    
    const handleSubmit = async () => {
        setLoading(true);
        await SubmitBudgetMaster(formData);
        fetchData();
        handleClose();
    };

    useEffect(() => {
        const updateBody = () => {
            setDetails(prevDetails => ({
                ...prevDetails,
                Body: <>
                    <div className="form-group">
                        <label htmlFor="budget_name" className="form-check-label">Enter your Budget Name:
                            <input
                                type="text"
                                name="budget_name"
                                id="budget_name"
                                value={formData.budget_name}
                                onChange={handleChange}
                                className="input-form-control"
                            />
                        </label>
                        
                    </div>
                </>
            }));
        };
        updateBody();
        // eslint-disable-next-line
    }, [formData]); 
    if (loading) {
        return <BusyIndicator />;
      }
    return (
        <>

            <div className="clipping-container">
                <Dialog DialogDetails={details}
                    onSubmit={handleSubmit}
                    onClose={handleClose}
                />
            </div>
        </>

    );
};

export default BudgetMasterForm;
