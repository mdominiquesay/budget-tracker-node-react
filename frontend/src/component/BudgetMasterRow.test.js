// BudgetMasterForm.test.js

import React from 'react';
import { render } from '@testing-library/react';
import BudgetMasterForm from './BudgetMasterForm'; // Update the import path as necessary

// Test the BudgetMasterForm component
describe('BudgetMasterForm', () => {
  it('renders correctly', () => {
    // Mock props
    const props = {
      budget_name: 'Test Budget',
      handleChange: jest.fn(),
    };

    // Render the component
    const { container } = render(<BudgetMasterForm {...props} />);

    // Assert that the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });
});
