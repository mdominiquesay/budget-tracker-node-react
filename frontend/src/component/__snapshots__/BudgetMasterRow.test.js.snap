// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetMasterForm renders correctly 1`] = `
<div>
  <div
    class="form-check mb-3"
  >
    <label
      class="form-check-label"
      for="budget_name"
    >
      Enter your Budget Name:
      <input
        class="form-control"
        id="budget_name"
        name="budget_name"
        type="text"
        value="Test Budget"
      />
    </label>
  </div>
</div>
`;
